packages/utils/node-resolver-core/src/NodeResolver.js

```diff
@@ -906,11 +906,14 @@ export default class NodeResolver {
     pkg.pkgfile = file;
     pkg.pkgdir = dir;
 
-    // If the package has a `source` field, check if it is behind a symlink.
-    // If so, we treat the module as source code rather than a pre-compiled module.
+    // If the package has a `source` field, make sure
+    // - the package is behind symlinks
+    // - and the realpath to the packages does not includes `node_modules`.
+    // Since such package is likely a pre-compiled module
+    // installed with package managers, rather than including a source code.
     if (pkg.source) {
       let realpath = await this.fs.realpath(file);
-      if (realpath === file) {
+      if (realpath === file || realpath.includes('/node_modules/')) {
```

Comment:
Let me know if this is not be a cross-platform approach..

Reply:
This is the only other place currently that does this kind of check (using `path.sep` instead of hardcoding the slash):
https://github.com/parcel-bundler/parcel/blob/96c847b21361bcd2f7c8712752e5376e1f80b27e/packages/core/core/src/summarizeRequest.js#L22-L24

Reply:
Okay, I just wondered `fs.realpath()` normalizes path separator or not. I once copied the snippet (9c9d379) you suggested, but it seemed not a good practice to paste the same constant here and there in the codebase, so revised with `normalizeSeparators()` (255ba5b).